pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -543,3 +567,20 @@ func constructFilesSecret(driver string, config map[string]interface{}) (*corev1
 	}
 	return nil, nil
 }
+
+func shouldForceRemoval(job *batchv1.Job, d data.Object) bool {
+	var createJobCondition summary.Condition
+	for _, c := range summary.GetUnstructuredConditions(d) {
+		if c.Type() == createJobConditionType {
+			createJobCondition = c
+			break
+		}
+	}
+
+	if createJobCondition.Reason() != string(errors.CreateMachineError) &&
```

Comment:
I take it that expected behavior here is that if the `createJobCondition.Reason()` is in fact the `CreateMachineError`, we should fall down to the bottom and return true if the failure threshold has been passed for the create job?

Reply:
Yes, that is correct.